nautobot/docs/release-notes/version-2.3.md

```diff
@@ -10,18 +10,34 @@ This document describes all new features and changes in Nautobot 2.3.
 
 Administrators should plan to take these actions during or immediately after upgrade from a previous version. New installations should also take note of these actions where appropriate.
 
+- Python 3.12 is now the default recommended version of Python.
+
+!!! warning "Python 3.12"
+    Because Nautobot prior to 2.3.0 did not declare support for Python 3.12, most Apps similarly needed to previously declare an upper bound of Python 3.11 for their own compatibility. Therefore, older versions of most Apps **will not be installable** under Python 3.12. Before migrating your Nautobot environment to Python 3.12, it is your responsibility to confirm that all relevant Apps in your environment are also compatible and installable.
+
+    There is a minor "chicken-and-egg" problem here in that Apps generally cannot declare support for a new Python version before Nautobot itself publishes a release that does so; therefore, as of the 2.3.0 Nautobot release day, most Apps have not yet been updated to declare support for Python 3.12. We'll be working in the following days to promptly update our supported Apps as needed, so stay tuned.
```

Comment:
Does this mean that Apps may need to drop support of Nautobot 2.2 because it's not compatible with 3.12?

I could see Poetry/pip complaining about resolving:
```
python = ">=3.8,<3.13"
nautobot = "^2.0.0"
```

Reply:
Appears fine for Poetry at least, it just resolves to nautobot 2.3.0b1 since that supports the full python range.

Reply:
LGTM